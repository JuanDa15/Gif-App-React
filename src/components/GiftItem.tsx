interface Props {
  url: string,
  width: string,
  height: string
}

export default function GiftItem ({url, width, height}: Props) {
  return (
    <li >
      <img  
        className="h-auto max-w-full rounded-lg aspect-square" 
        src={url} 
        width={width}
        height={height}
        loading='lazy'
      />
    </li>
  )
}