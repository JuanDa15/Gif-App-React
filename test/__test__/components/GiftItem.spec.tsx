import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import GiftItem from '../../../src/components/GiftItem'

describe('Test in <GiftItem />', () => {
  const props = {
    url: 'https://picsum.photos/200/300',
    width: '200',
    height: '300'
  }
  test("Renders the component", () => {
    render(<GiftItem {...props} />)
    expect(true).toBeTruthy()
  })
  test("should match with snapshot", () => {
    const { container } = render(<GiftItem {...props} />)
    expect(container).toMatchSnapshot()
  })
  test("should have all the attributes", () => {
    render(<GiftItem {...props} />)
    const { src, width, height} = screen.getByRole('img') as HTMLImageElement;
    expect(src).toBe(props.url)
    expect(width.toString()).toEqual(props.width)
    expect(height.toString()).toEqual(props.height)
  })
})