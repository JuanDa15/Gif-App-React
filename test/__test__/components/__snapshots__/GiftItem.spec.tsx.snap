// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GiftItem /> should match with snapshot 1`] = `
<div>
  <li>
    <img
      class="h-auto max-w-full rounded-lg aspect-square"
      height="300"
      loading="lazy"
      src="https://picsum.photos/200/300"
      width="200"
    />
  </li>
</div>
`;
